esmvalcore/_main.py

```diff
@@ -392,13 +392,28 @@ def run(self,
 
         self._run(recipe, session)
 
+    @staticmethod
+    def _create_session_dir(session):
+        """Create `session.session_dir` or an alternative if it exists."""
+        from .exceptions import RecipeError
+
+        session_dir = session.session_dir
+        for suffix in range(1, 1000):
+            try:
+                session_dir.mkdir(parents=True)
+            except FileExistsError:
+                session_dir = Path(f"{session.session_dir}-{suffix}")
+            else:
+                session.session_name = session_dir.name
```

Comment:
Is it necessary to overwrite `session.session_dir` at this point as well?

Reply:
That is not needed because it's computed from `Session.session_name`: https://github.com/ESMValGroup/ESMValCore/blob/96419a7e8d3321e7d07b33f390b924ebced8e102/esmvalcore/config/_config_object.py#L169-L172